projects/ngrx.io/content/guide/effects/index.md

```diff
@@ -218,3 +218,103 @@ export class MovieModule {}
 
 </div>
+
+## Incorporating State
+
+If additional metadata is needed to perform an effect besides the initiating action's `type`, we should rely on passed metadata from an action creator's `props` method.
+
+Let's look at an example of an action initiating a login request using an effect with additional passed metadata:
+
+<code-example header="login-page.actions.ts">
+import { createAction, props } from '@ngrx/store';
+import { Credentials } from '@example-app/auth/models/user';
+
+export const login = createAction(
+  '[Login Page] Login',
+  props&lt;{ credentials: Credentials }&gt;()
+);
+
+export type LoginPageActionsUnion = ReturnType&lt;typeof login&gt;;
+</code-example>
+
+<code-example header="auth.effects.ts">
+import { Injectable } from '@angular/core';
+import { Actions, ofType, createEffect } from '@ngrx/effects';
+import { of } from 'rxjs';
+import { catchError, exhaustMap, map } from 'rxjs/operators';
+import {
+  LoginPageActions,
+  AuthApiActions,
+} from '@example-app/auth/actions';
+import { Credentials } from '@example-app/auth/models/user';
+import { AuthService } from '@example-app/auth/services/auth.service';
+
+@Injectable()
+export class AuthEffects {
+  login$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(LoginPageActions.login),
+      map(action => action.credentials),
+      exhaustMap((auth: Credentials) =>
+        this.authService.login(auth).pipe(
+          map(user => AuthApiActions.loginSuccess({ user })),
+          catchError(error => of(AuthApiActions.loginFailure({ error })))
+        )
+      )
+    )
+  );
+
+  constructor(
+    private actions$: Actions,
+    private authService: AuthService
+  ) {}
+}
+</code-example>
+
+The `login` action has additional `credentials` metadata which is passed to a service to log the specific user into the application.
+
+However, there may be cases when the required metadata is only accessible from state.  When state is needed, the RxJS `withLatestFrom` operator can be used to provide it.
+
+The example below shows the `addBookToCollectionSuccess$` effect displaying a different alert depending on the number of books in the collection state.
+
+<code-example header="collection.effects.ts">
+import { Injectable } from '@angular/core';
+import { Store, select } from '@ngrx/store';
+import { Actions, ofType, createEffect } from '@ngrx/effects';
+import { of } from 'rxjs';
+import { map, concatMap, withLatestFrom } from 'rxjs/operators';
+import { CollectionApiActions } from '@example-app/books/actions';
+import * as fromBooks from '@example-app/books/reducers';
+
+@Injectable()
+export class CollectionEffects {
+  addBookToCollectionSuccess$ = createEffect(
+    () =>
+      this.actions$.pipe(
+        ofType(CollectionApiActions.addBookSuccess),
+        concatMap(action => of(action).pipe(
+          withLatestFrom(this.store.pipe(select(fromBooks.getCollectionBookIds)))
```

Comment:
Another preference here, what about using the RxJS `project` function here.

```ts
withLatestFrom(this.store.pipe(select(fromBooks.getCollectionBookIds)), ([_action, collection]) => collection)
```

Reply:
The recommended way is to use the map instead of the projector functions with most of the other operators, so I don't see it being an issue here.

Reply:
Definitely could.  In my opinion, it complicates the example.